swift/Tests/Blockchains/HarmonyTests.swift

```diff
@@ -30,4 +31,178 @@ class HarmonyTests: XCTestCase {
         XCTAssertEqual(output.r.hexString, "325aed6caa01a5235b7a508c8ab67f0c43946b05a1ea6a3e0628de4033fe372d")
         XCTAssertEqual(output.s.hexString, "6c19085d3376c30f6dc47cec795991cd37d6d0ebddfa633b0a8f494bc19cd01b")
     }
-}
+
+    func testStakingCreateValidator() {
+        let description = TW_Harmony_Proto_Description.with {
+            $0.name = "Alice"
+            $0.identity = "alice"
+            $0.website = "alice.harmony.one"
+            $0.securityContact = "Bob"
+            $0.details = "Don't mess with me!!!"
+        }
+        let rate = TW_Harmony_Proto_Decimal.with {
+            $0.value = "0.1"
+        }
+        let maxRate = TW_Harmony_Proto_Decimal.with {
+            $0.value = "0.9"
+        }
+        let maxChangeRate = TW_Harmony_Proto_Decimal.with {
+            $0.value = "0.05"
+        }
+        let commission = TW_Harmony_Proto_CommissionRate.with {
+            $0.rate = rate
+            $0.maxRate = maxRate
+            $0.maxChangeRate = maxChangeRate
+        }
+        let pubKey = Data(hexString: "b9486167ab9087ab818dc4ce026edb5bf216863364c32e42df2af03c5ced1ad181e7d12f0e6dd5307a73b62247608611")!
+        let createValidator = TW_Harmony_Proto_DirectiveCreateValidator.with {
+            $0.validatorAddress = "one1a0x3d6xpmr6f8wsyaxd9v36pytvp48zckswvv9"
+            $0.description_p = description
+            $0.commissionRates = commission
+            $0.minSelfDelegation = Data(hexString: "0xa")!
+            $0.maxTotalDelegation = Data(hexString: "0x0bb8")!
+            $0.slotPubKeys = [pubKey]
+            $0.amount = Data(hexString: "0x64")!
+        }
+        let input = TW_Harmony_Proto_StakingTransactionInput.with {
+            $0.chainID = Data(hexString: localNet)!
+            $0.privateKey = Data(hexString: "4edef2c24995d15b0e25cbd152fb0e2c05d3b79b9c2afd134e6f59f91bf99e48")!
+            $0.nonce = Data(hexString: "0x2")!
+            $0.gasPrice = Data(hexString: "0x")!
+            $0.gasLimit = Data(hexString: "0x64")!
+            $0.createValidatorMessage = createValidator
+        }
+        let output = HarmonyStakingSigner.signCreateValidator(input: input)
+
+        let e1 = "f8ed80f8a494ebcd16e8c1d8f493ba04e99a56474122d81a9c58f83885416c69636585616c69636591616c6963"
+        let e2 = "652e6861726d6f6e792e6f6e6583426f6295446f6e2774206d6573732077697468206d65212121ddc988016345"
+        let e3 = "785d8a0000c9880c7d713b49da0000c887b1a2bc2ec500000a820bb8f1b0b9486167ab9087ab818dc4ce026edb"
+        let e4 = "5bf216863364c32e42df2af03c5ced1ad181e7d12f0e6dd5307a73b622476086116402806428a0476e8a0fe478"
+        let e5 = "e0d03ff10222d4d590bca8cee3ec51b830f4fc4a8bee5d0e9d28a03b2be18e73b2f99d7e2691485a0e166f28e6"
+        let e6 = "2815079c126e68f876dc97339f8f"
+
+        XCTAssertEqual(output.encoded.hexString, e1 + e2 + e3 + e4 + e5 + e6)
```

Comment:
what's the point of e1~e6?

Reply:
Just split the long string into multiple smaller ones (automatically done by the code formatting). Do you need it as a one long string?

Reply:
Made them into single long string.